File: android/app/src/_01/java/com/elliott/sxswreduxworkshopandroid/redux/AppReducer.kt
package com.elliott.sxswreduxworkshopandroid.redux

import redux.api.Action
import redux.api.Reducer

class AppReducer : Reducer<AppState> {

    override fun reduce(state: AppState, action: Action?): AppState {
        return when (action) {
            is ListStateReducer -> {
                val listState = action.reduce(state.listState)
                state.copy(listState = listState)
            }
            else -> state
        }
    }
}

File: ios/ex1_SearchValidation/NASARedux/Modules/Application/Redux/AppReducer.swift
//
//  AppReducer.swift
//  NASARedux
//
//  Created by John McIntosh on 2/23/19.
//  Copyright © 2019 Demo. All rights reserved.
//

import Foundation
import ReSwift

func appReducer(action: Action, state: AppState?) -> AppState {
    var state = state ?? AppState.makeDefault()
    
    if let action = action as? ListReducible {
        action.reduce(state: &state.listState)
    }
    
    return state
}

private extension AppState {
    
    static func makeDefault() -> AppState {
        return AppState(listState: ListState())
    }
}
